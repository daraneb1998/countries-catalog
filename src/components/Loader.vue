<script setup lang="ts">
import { useSearchStore } from "@/stores/search";
import VueLoading from 'vue-loading-overlay';
import "vue-loading-overlay/dist/css/index.css";

const searchStore = useSearchStore();
</script>

<template>
  <div class="vl-parent">
    <VueLoading
      :visible="searchStore.isLoading"
      :color="'#4fc08d'"
      :size="'large'"
    ></VueLoading>
  </div>
</template>
