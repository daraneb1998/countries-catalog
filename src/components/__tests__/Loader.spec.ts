import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import Loader from "../Loader.vue";
import { createPinia, setActivePinia } from "pinia";

describe("Loader", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it("renders properly", () => {
    const wrapper = mount(Loader);
    expect(wrapper.text()).toMatchSnapshot();
  });
});
