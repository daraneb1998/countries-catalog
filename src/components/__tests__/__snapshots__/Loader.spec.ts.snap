// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconSearch > renders properly 1`] = `""`;

exports[`Loader > renders properly 1`] = `""`;
